utils/serde/src/main/java/io/apicurio/registry/utils/serde/JsonSchemaKafkaSerializer.java

```diff
@@ -0,0 +1,192 @@
+     * @param client
+     */
+    public JsonSchemaKafkaSerializer(RegistryService client) {
+        super(client);
+    }
+
+    /**
+     * @see org.apache.kafka.common.serialization.Serializer#configure(java.util.Map, boolean)
+     */
+    @Override
+    public void configure(Map<String, ?> configs, boolean isKey) {
+        super.configure(configs);
+        
+        Object ve = configs.get(REGISTRY_JSON_SCHEMA_SERIALIZER_VALIDATION_ENABLED);
+        this.validationEnabled = ve != null && ("true".equals(ve) || ve.equals(Boolean.TRUE));
+    }
+    
+    /**
+     * @see org.apache.kafka.common.serialization.Serializer#serialize(java.lang.String, java.lang.Object)
+     */
+    @Override
+    public byte[] serialize(String topic, T data) {
+        // Headers are required when sending data using this serdes impl
+        throw new UnsupportedOperationException();
+    }
+    
+    /**
+     * @see org.apache.kafka.common.serialization.Serializer#serialize(java.lang.String, org.apache.kafka.common.header.Headers, java.lang.Object)
+     */
+    @Override
+    public byte[] serialize(String topic, Headers headers, T data) {
+        if (data == null) {
+            return new byte[0];
+        }
+
+        // Now serialize the data
+        try {
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            JsonGenerator generator = mapper.getFactory().createGenerator(baos);
+            if (validationEnabled) {
+                String artifactId = getArtifactId(topic, data);
+                Integer version = getArtifactVersion(artifactId, topic, data);
+                addSchemaHeaders(headers, artifactId, version);
+
+                String schema = loadSchema(artifactId, version);
+                SchemaValidator schemaValidator = api.loadSchema(new StringSchemaSource(schema));
+                // TODO cache the SchemaValidator - keyed on artifactId+version
+                generator = api.decorateJsonGenerator(schemaValidator, generator);
+            }
+            addTypeHeaders(headers, data);
+
+            mapper.writeValue(generator, data);
+            
+            return baos.toByteArray();
+        } catch (IOException e) {
+            throw new SerializationException(e);
+        }
+    }
+
+    /**
+     * Figure out the artifact ID from the topic name and data.
+     * @param topic
+     * @param data
+     */
+    private String getArtifactId(String topic, T data) {
+        // TODO support other options - other impls use a strategy for this
+        return topic;
+    }
+
+    /**
+     * Figure out which version of the artifact to use, specifically.  If this returns
+     * null then the latest version will be used.
+     * @param artifactId
+     * @param topic
+     * @param data
+     */
+    private Integer getArtifactVersion(String artifactId, String topic, T data) {
+        // TODO how can we know what artifact version we need?  passed in via config?
+        return null;
```

Comment:
Use GlobalIdStrategy ... and don't return version ... but globalId.
So we have same behavior for all serializers ...